### ISSUE TITLE
`\Magento\ImportExport\Block\Adminhtml\Export\Filter::_getSelectHtmlWithValue()` method overwrites self $value argument

### ISSUE THREAD

**Issue by andkirby** [magento/magento2#20624]
There is the issue described in the summary.

### Preconditions (*)

* Magento EE v2.2.5
* Targed class/method is `\Magento\ImportExport\Block\Adminhtml\Export\Filter::_getSelectHtmlWithValue()`

### Steps to reproduce (*)
It's clear in the code the `$value` variable
```php
protected function _getSelectHtmlWithValue(Attribute $attribute, $value)
```
is gonna be overwritten in
```php
            foreach ($attribute->getFilterOptions() as $value => $label) {
                $options[] = ['value' => $value, 'label' => $label];
            }
```

### Expected result (*)
Different names for different `$value` variables must be set.

### Actual result (*)
There is same name for different `$value` variables.

**Comment by magento-engcom-team** [comment]
Hi @andkirby. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [x] Summary of the issue
- [x] Information on your environment
- [x] Steps to reproduce
- [x] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@andkirby do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by andkirby** [comment]
Yeah, that's EE issue. Sorry, I don't have an account to a special place of EE issues. You may grant me. :)

**Comment by magento-engcom-team** [comment]
Hi @irajneeshgupta. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
@irajneeshgupta Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-97935`, `MAGETWO-97936` were created

**Comment by magento-engcom-team** [comment]
Hi @Nazar65. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by irajneeshgupta** [comment]
@andkirby 
You reported issue for EE , it is actually in CE version also , hence i confirmed it.
For EE version you should open ticket on EE cloud support

**Comment by andkirby** [comment]
> @andkirby
> You reported issue for EE , it is actually in CE version also , hence i confirmed it.
> For EE version you should open ticket on EE cloud support

Got it, thanks. I just don't have access to this EE cloud support.

**Comment by andkirby** [comment]
> @andkirby
> You reported issue for EE , it is actually in CE version also , hence i confirmed it.
> For EE version you should open ticket on EE cloud support

@irajneeshgupta Yeah, I just confused `magento/module-scheduled-import-export` from EE with `magento/module-import-export` which belongs to CE as well. So, that's CE issue basically.

**Comment by andkirby** [comment]
@irajneeshgupta I've added a [suggestion](https://github.com/irajneeshgupta/magento2/commit/7e4691723e6e3118404541c8e7625346dd8e8545#comments) for variable name. Could you please apply it?

**Comment by irajneeshgupta** [comment]
@andkirby 
I added your suggestion , it will be applied when PR will merged.

**Comment by magento-engcom-team** [comment]
Hi @swnsma. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by magento-engcom-team** [comment]
Hi @andkirby. Thank you for your report. 
The issue has been fixed in magento/magento2#20625 by @irajneeshgupta in 2.3-develop branch
Related commit(s):
 - [7e4691723e6e3118404541c8e7625346dd8e8545](https://github.com/magento/magento2/commit/7e4691723e6e3118404541c8e7625346dd8e8545)
 - [de2010cedd15a9a5d8c0bc8703e3af87b849eeec](https://github.com/magento/magento2/commit/de2010cedd15a9a5d8c0bc8703e3af87b849eeec)

The fix will be available with the upcoming 2.3.1 release.

**Comment by magento-engcom-team** [comment]
Hi @andkirby. Thank you for your report. 
The issue has been fixed in magento/magento2#20863 by @irajneeshgupta in 2.2-develop branch
Related commit(s):
 - [05bc8a603a563bf54601d11e5af5f46eb9b6a716](https://github.com/magento/magento2/commit/05bc8a603a563bf54601d11e5af5f46eb9b6a716)
 - [34da4ca890562dc90d7f4f9610cafd79d2694f7b](https://github.com/magento/magento2/commit/34da4ca890562dc90d7f4f9610cafd79d2694f7b)

The fix will be available with the upcoming 2.2.8 release.
